### ISSUE TITLE
The replace function throws error for unmatched subgroups

### ISSUE THREAD

**Issue by jaschiu** [khchen/tinyre#1]
If the regex contains subgroups that never matched, then their boundary slice would be from -1 to -1.

My regex: `\$(?:(\$)|([_a-z]\w*)|\{([_a-z]\w*)(?:(:?)([-\?])([^\}]*))?\})`

Error:
```
/home/lj/.local/share/nimble/pkgs2/tinyre-1.5.1-60bc507bc3d3fe395c4d6444c0de1880ddcfa893/tinyre.nim(430) replace
/home/lj/.choosenim/toolchains/nim-2.0.0/lib/system/indices.nim(86) []
/home/lj/.choosenim/toolchains/nim-2.0.0/lib/system/fatal.nim(53) sysFatal
Error: unhandled exception: index -1 not in 0 .. 16 [IndexDefect]
```

**Comment by khchen** [comment]
Can you provide the full code that raises the error?

**Comment by jaschiu** [comment]
Unfortunately, I do not have the code that can reproduce the error as I switched to a different regex library and this was many months ago. Although it seems that this regex tries to emulate dollar-sign environment variable replacement like `envsubst`. I'll close this for now

**Comment by khchen** [comment]
I believe this issue has been fixed in version 1.6.0.
